### PR TITLE
snapcraft: add python-uefivars

### PR DESCRIPTION
**Pull request by mihalicyn**
We need it for "lxc config uefi <...>" extension

There is some trickiness around this thing, because for some reason snapcraft is going crazy and everything that builds after "uefivars" starts to fail with some Python errors (failed to find package XYZ).

I have workarounded it by adding "after" specifier for uefivars part.

### CODE REVIEW

**Comment by mihalicyn** [comment]
I have no posted a PR for LXD yet, but I'll do that in next few days. It's safe to merge this before that as here we are just adding a new component.

File: snapcraft.yaml [hunk 1]
```diff
@@ -1375,6 +1376,16 @@ parts:
 
       patch -p1 "${CRAFT_PART_INSTALL}/snap/lxd/current/lxcfs/lxc.mount.hook" < "${CRAFT_PROJECT_DIR}/patches/lxcfs-0001-hook.patch"
 
+  uefivars:
```
Comment (MusicDin) [inline snapcraft.yaml:1381]: I see there is only one release of python-uefivars - but should we bind it to the specific tag (`1.0.0`), or is latest ok?

Comment (tomponline) [reply snapcraft.yaml:1381]: we should indeed or a git sha

**Review by simondeziel — COMMENTED** [review]
Agreed on the version pinning suggested by Din.

As for this new part, it should be noted that once on core24, we will be able to use the [`python3-uefivars`](https://packages.ubuntu.com/noble/python3-pyuefivars) package instead :)

File: snapcraft.yaml (hunk 1 above)
Comment (mihalicyn) [reply snapcraft.yaml:1381]: Sure, I'll do version pinning. My bad. Thanks!

File: snapcraft.yaml [hunk 2]
```diff
@@ -600,6 +600,7 @@ parts:
     autotools-configure-parameters:
       - --prefix=
       - --with-json
+      - --disable-man-doc
```
Comment (tomponline) [inline snapcraft.yaml:605]: This should be a separate commit to make back porting easier please

File: snapcraft.yaml (hunk 1 above)
Comment (tomponline) [reply snapcraft.yaml:1381]: @mihalicyn please can you do the version pinning and i can merge this, ta

File: snapcraft.yaml (hunk 2 above)
Comment (mihalicyn) [reply snapcraft.yaml:605]: Have done!

File: snapcraft.yaml (hunk 1 above)
Comment (mihalicyn) [reply snapcraft.yaml:1381]: Have done!

**Review by tomponline — APPROVED** [review]
Thanks!
